platforms/software/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -1084,21 +1107,37 @@ public void maven(String comment, String url) {
     }
 
     private static class DependenciesBlock implements DependenciesBuilder, Statement, BlockBody {
+        final BuildScriptBuilder buildScriptBuilder;
         final ListMultimap<String, Statement> dependencies = MultimapBuilder.linkedHashKeys().arrayListValues().build();
         final ListMultimap<String, Statement> constraints = MultimapBuilder.linkedHashKeys().arrayListValues().build();
 
+        public DependenciesBlock(BuildScriptBuilder buildScriptBuilder) {
```

Comment:
💅 I would only pass the version catalog registry here instead of the whole build script builder

Reply:
With the latest changes 2 members are required and passing the complete `BuildScriptBuilder` is consistent with other blocks